zulipterminal/ui_tools/views.py

```diff
@@ -589,3 +590,107 @@ def keypress(self, size: Tuple[int, int], key: str) -> str:
         if is_command_key('GO_BACK', key) or is_command_key('HELP', key):
             self.controller.exit_pop_up()
         return super(HelpView, self).keypress(size, key)
+
+
+class MsgInfoView(urwid.ListBox):
+    def __init__(self, controller: Any, msg: Any) -> None:
+        self.controller = controller
+        self.msg = msg
+
+        if msg['type'] == 'stream':
+            recipient = msg['display_recipient'] + " -> " + msg['subject']
+        elif not msg['display_recipient'][0]['full_name'] ==\
+                msg['sender_full_name']:
+            recipient = msg['display_recipient'][0]['full_name']
+        else:
+            recipient = 'Self'
+
+        msg_info = OrderedDict({
+            'Date & Time': time.ctime(msg['timestamp'])[:-5],
+            'Sender': msg['sender_full_name'],
+            'Sender\'s Email ID': msg['sender_email'],
+            'Recipient': recipient,
+            'Reactions': sorted(
+                    [reaction['emoji_name'] +
+                        ": " +
+                        reaction['user']['full_name'] +
+                        "\n"
+                        for reaction in msg['reactions']])
+            if msg['reactions']
+            else '---None---',
+            'Edit History': 'Press \'e\' to view',
+            })
+
+        widths = [(len(field[0])+7,
+                  max(len(reaction_users) for reaction_users in field[1])
+                  if isinstance(field[1], list)
+                  else len(field[1])+2)
+                  for field in msg_info.items()]
+        max_widths = [max(width) for width in zip(*widths)]
+        self.width = sum(max_widths)
+        self.height = min(len(msg_info['Reactions'])+6 if msg['reactions']
+                          else 6, 27)
+
+        self.log = urwid.SimpleListWalker(
+            [urwid.AttrWrap(
+                urwid.Columns([
+                    urwid.Text(field[0]),
+                    (max_widths[1], urwid.Text(field[1]))
+                ], dividechars=2),
+                None if index % 2 else 'bar')
+             for index, field in enumerate(msg_info.items())])
```

Comment:
Maybe something like the following would simplify this further? (or other variable names, of course)
`for index, (title, data) in enumerate(msg_info.items()`